odoo/addons/base/models/ir_ui_view.py

```diff
@@ -2204,7 +2205,12 @@ def __init__(self, model):
 
     @lazy_property
     def field_info(self):
-        return self.model.fields_get()
+        field_info = self.model.fields_get()
+        has_access = functools.partial(self.model.check_access_rights, raise_exception=False)
+        if not (has_access('write') or has_access('create')):
+            for info in field_info.vals():
+                info.update({'readonly': True, 'states': {}})
```

Comment:
:disappointed:

Reply:
??

Reply:
`update` is a waste of resources for so few keys, and creating a literal dict even more so. The original code actually did it "the right way".

Reply:
solved